sources/api/apiclient/src/uri_resolver.rs

```diff
@@ -0,0 +1,884 @@
+            .region(aws_sdk_ssm::config::Region::new(self.region.clone()))
+            .load()
+            .await;
+
+        let client = aws_sdk_ssm::Client::new(&cfg);
+
+        // 2) Fetch the parameter with decryption
+        let resp = client
+            .get_parameter()
+            .name(self.parameter_name.clone())
+            .with_decryption(true)
+            .send()
+            .await
+            .context(SsmGetParameterSnafu {
+                parameter_name: self.parameter_name.clone(),
+            })?;
+
+        // 3) Extract the string value
+        let value = resp
+            .parameter
+            .and_then(|p| p.value().map(|v| v.to_string()))
+            .context(SsmParameterMissingSnafu {
+                parameter_name: self.parameter_name.clone(),
+            })?;
+
+        Ok(value)
+    }
+}
+
+/// Uri Resolver that fetches parameters from AWS SSM Parameter Store.
+///
+/// This resolver accepts URIs of the form `ssm://parameter_name`, uses the
+/// AWS SDK’s default credential and region resolution, and returns the value
+/// of the requested parameter.
+pub struct SsmUri {
+    parameter_name: String,
+}
+
+impl TryFrom<&SettingsInput> for SsmUri {
+    type Error = ResolverError;
+    fn try_from(input: &SettingsInput) -> ResolverResult<Self> {
+        use resolver_error::*;
+
+        const PREFIX: &str = "ssm://";
+        let uri_str = input.input.as_str();
+        let remainder = uri_str.strip_prefix(PREFIX).context(SsmUriSnafu {
+            input_source: input.input.clone(),
+        })?;
+        ensure!(
+            !remainder.is_empty(),
+            SsmUriSnafu {
+                input_source: input.input.clone()
+            }
+        );
+
+        Ok(SsmUri {
+            parameter_name: remainder.to_string(),
+        })
+    }
+}
+
+#[async_trait]
+impl UriResolver for SsmUri {
+    async fn resolve(&self) -> ResolverResult<String> {
+        // use default region chain
+        let config = aws_config::load_defaults(aws_config::BehaviorVersion::latest()).await;
+        let client = ssm::Client::new(&config);
+        use resolver_error::*;
+
+        // fetch the parameter, with decryption
+        let resp = client
+            .get_parameter()
+            .name(self.parameter_name.clone())
+            .with_decryption(true)
+            .send()
+            .await
+            .context(SsmGetParameterSnafu {
+                parameter_name: self.parameter_name.clone(),
+            })?;
+
+        // extract the string value
+        let value = resp
+            .parameter
+            .and_then(|p| p.value().map(|v| v.to_string()))
+            .context(SsmParameterMissingSnafu {
+                parameter_name: self.parameter_name.clone(),
+            })?;
+
+        Ok(value)
+    }
+}
+
+#[derive(Debug, Snafu)]
+#[snafu(module)]
+pub enum ResolverError {
+    #[snafu(display("Invalid ARN '{}': {}", input_source, reason))]
+    InvalidArnFormat {
```

Comment:
I'd prefer having this be composed of smaller Resolver-specific types like `FileUriResolverError`, `S3ResolverError`, `SsmResolverError`, etc would improve maintainability.